src/77480.c

```diff
@@ -848,7 +904,90 @@ void appendGfx_player(void) {
 }
 
 /// Only used when speedy spinning.
-INCLUDE_ASM(s32, "77480", appendGfx_player_spin);
+void appendGfx_player_spin(void) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Matrix4f mtx;
+    Matrix4f translation;
+    Matrix4f rotation;
+    Matrix4f scale;
+    f32 yaw;
+    f32 blurAngle;
+    s32 tint;
+    f32 px, py, pz;
+    s32 x, y, z;
+    s32 i;
+    s32 flags;
+
+    for (i = 0; i < 2; i++) {
+        yaw = -gCameras[gCurrentCamID].currentYaw;
+
+        if (i == 0) {
+            if ((playerStatus->spriteFacingAngle > 90.0f) && (playerStatus->spriteFacingAngle <= 180.0f)) {
+                yaw = 180.0f - playerStatus->spriteFacingAngle;
+            } else {
+                if ((playerStatus->spriteFacingAngle > 180.0f) && (playerStatus->spriteFacingAngle <= 270.0f)) {
+                    yaw = playerStatus->spriteFacingAngle - 180.0f;
+                } else if ((playerStatus->spriteFacingAngle > 270.0f) && (playerStatus->spriteFacingAngle <= 360.0f)) {
+                    yaw = 360.0f - playerStatus->spriteFacingAngle;
+                } else {
+                    yaw = playerStatus->spriteFacingAngle;
+                }
+            }
+
+            tint = yaw / 25.0f;
+            tint = 255 - (tint * 60);
+            if (tint < 100) {
+                tint = 100;
+            }
+
+            func_802DDFF8(0, 6, tint, tint, tint, 255, 0);
+
+            guRotateF(rotation, yaw, 0.0f, -1.0f, 0.0f);
+            guRotateF(mtx, clamp_angle(playerStatus->unk_8C), 0.0f, 0.0f, 1.0f);
+            guMtxCatF(rotation, mtx, mtx);
+            px = playerStatus->position.x;
+            py = playerStatus->position.y;
+            pz = playerStatus->position.z;
+        } else {
+            blurAngle = phys_get_spin_history(i, &x, &y, &z);
+
+            if (y == 0x80000000) {
+                py = playerStatus->position.y;
+            } else {
+                py = y;
+            }
+
+            px = playerStatus->position.x;
+            pz = playerStatus->position.z;
+            func_802DDEE4(0, -1, 7, 0, 0, 0, 0x40, 0);
+            guRotateF(mtx, yaw, 0.0f, -1.0f, 0.0f);
+            guRotateF(rotation, yaw, 0.0f, -1.0f, 0.0f);
+            guRotateF(mtx, blurAngle, 0.0f, 1.0f, 0.0f);
+            guMtxCatF(rotation, mtx, mtx);
+        }
+
+        guTranslateF(translation, 0.0f, -playerStatus->colliderHeight * 0.5f, 0.0f);
+        guMtxCatF(translation, mtx, mtx);
+        guRotateF(rotation, yaw, 0.0f, 1.0f, 0.0f);
+        guMtxCatF(mtx, rotation, mtx);
+        guRotateF(rotation, playerStatus->spriteFacingAngle, 0.0f, 1.0f, 0.0f);
+        guMtxCatF(mtx, rotation, mtx);
+        guTranslateF(translation, 0.0f, playerStatus->colliderHeight * 0.5f, 0.0f);
+        guMtxCatF(mtx, translation, mtx);
+        guScaleF(scale, 0.71428573f, 0.71428573f, 0.71428573f);
```

Comment:
0.71428573f is `5.0/7.0` or something - see #580

Reply:
This needs to be done as part of the merge I think

Reply:
Yeah, this is `SPRITE_PIXEL_SCALE` now, so you probably need to rebase, otherwise you introduce this faulty into the codebase